File: checkIfTwoStringArraysAreEquivalent.py
from typing import List

class Solution:
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
    
        wordResult1List = []
        wordResult2List = []
        
        for i in range(0, len(word1)):
           wordResult1List.insert(len(wordResult1List),word1[i])
        
        for i in range(0, len(word2)):
            wordResult2List.insert(len(wordResult2List),word2[i])
        
        wordResult1 = "".join(wordResult1List)
        wordResult2 = "".join(wordResult2List)
        
        if (wordResult1 == wordResult2):
            result = 1
        
        else:
            result = 0
    
        return result

word1 = ["a", "cb"]
word2 = ["ab", "c"]

myObj = Solution()

answer = myObj.arrayStringsAreEqual(word1, word2)

print(answer)

File: checkIfTwoStringArraysAreEquivalent.cpp
#include <iostream>
using namespace std;

#include <vector>

class Solution {
public:
    bool arrayStringsAreEqual(vector<string>& word1, vector<string>& word2) 
    {
        string wordResult1;
        string wordResult2;
        
        bool result;
        
        for (int i = 0; i < word1.size(); i++)
        {
           wordResult1.append(word1[i]);
        }
        
        for (int i = 0; i < word2.size(); i++)
        {
            wordResult2.append(word2[i]);
        }
        
        if (wordResult1 == wordResult2)
        {
            result = true;
        }
        
        else
        {
            result = false;
        }
        
        return result;
    }
};

int main()
{
	vector<string> word1 = {"a", "cb"};
	vector<string> word2 = {"ab", "c"};

	Solution myObj;
	bool answer = myObj.arrayStringsAreEqual(word1, word2);

	cout << answer << endl; 

	return 0;
}
